app/src/main/java/com/spoony/spoony/presentation/report/ReportViewModel.kt

```diff
@@ -58,13 +50,21 @@ class ReportViewModel @Inject constructor(
         return input.length in minLength..maxLength
     }
 
-    fun reportPost(postId: Int, reportType: String, reportDetail: String) {
-        viewModelScope.launch {
-            reportRepository.postReportPost(postId = postId, userId = 4, reportType = reportType, reportDetail = reportDetail)
-                .onSuccess {
-                    _sideEffect.emit(ReportSideEffect.ShowDialog)
+    fun reportPost(reportType: String, reportDetail: String) {
+        val reportArgs = savedStateHandle.toRoute<Report>()
+        val postId = reportArgs.postId
+        val userId = reportArgs.userId
```

Comment:
P3: 오호 여기에서 postId와 userId를 받아오는걸로 수정하셨네요!! 화면에 진입했을 때가 아닌 버튼을 눌렀을 때 받아오도록 구현하신 이유가 무엇인가요??

Reply:
따로 postId, userId 를 UI 화면 상에서는 필요한 데이터가 아니라 판단되어서 화면 진입 시 저장하지 않고 
버튼 클릭했을 때 그때 viewModel 에서 알고 있는 postId, userId 를 사용하는게 더 효율적이라 생각했습니다.

Reply:
간단하게는 state 에 저장할 필요가 없다고 생각해서 변경한 코드입니다.

Reply:
저는 postId, userId를 받아오는 부분은 게시물을 신고하는 함수에 대한 역할이 아니라고 생각해서 데이터를 받아오는 부분은 viewModel의 init 함수 안에서 진행할 것 같습니다. 하지만 세홍님께서 말씀하신대로 해당 정보들이 ui에 필요한 데이터가 아니기 때문에 지금 방식대로 state에 저장하지 않고 진행하는 것도 좋은 방법인 것 같아요!!

추가적으로 postId, userId를 한 번 더 value로 저장하신 이유가 있으신가요?? 특별한 이유가 없다면 해당 변수들은 함수 내에서만 사용하기 때문에 아래와 같은 방식을 사용하는 것도 좋아보여요!
 
```suggestion
        val reportArgs = savedStateHandle.toRoute<Report>()
        with(reportArgs) { ... }
```

Reply:
오 좋네요 감사합니다!! 해당 부분으로 수정하겠습니다! 좋은 의견 감사해요!